### DISCUSSION TITLE
How can I edit the trailers metadata?

### DISCUSSION THREAD

**Discussion by OMGButterbee** [CherryFloors/jellyfin-plugin-cinemamode#19] [Q&A]
I've got a number of trailers that came with discs that I've put into the trailers subfolder for each movie. But the web interface doesn't let me view them so I have no way to edit them. I don't know how this plugin is determining the genre or year for the trailers. Is that just coming from the parent movie? Because sometimes the genre of the trailers aren't quite the same as the feature.

**Comment by CherryFloors** [comment]
>Is that just coming from the parent movie?

Yes, the plugin assumes that trailers stored with a movie are trailers for that movie.  When the plugin queries for trailers, it actually does a query for "Movies" and then filters based on the "Movies" number of trailers and metadata. I do believe this is the intention Jellyfin has for the "trailer" extra type. These are what will play when you press the "Trailer" button in the web UI. Without these local files, this button would link to trailers for the movie from youtube.

My intention for the plugin was  to show trailers for movies that were available on the server. If having trailers play for movies that you dont have on the server is something you are interested in, I can suggest a workaround:
1. Create a new library folder and in it, make a movie directory for every trailer you have (that doesn't have a matching movie in your library):
```
Trailer_Library 
    | - Movie_Name
        | -  Movie_Name.mp4
        | -  trailer.mp4
```
2. The file `trailer.ext` should be the trailer for "Movie_Name". To save space you could have "Movie_Name.mp4" be a symbolic link to "trailer.mp4" (linux) or a blank file text file renamed to "Movie_Name.mp4" (blank files will result in a playback error if anyone tries to play the actual movie).
3. Create a new library with type "Movies" pointing to this new directory. Jellyfin should scrape all the metadata for these "mock" movies with real trailers. When the plugin queries for movies with trailers, these will show up in the list and be filtered based on the metadata.

**Comment by OMGButterbee** [comment]
But then these mock movies would show up as full movies to the user right? Similar to how the bumpers do? So would have to make it very clear that these were trailers and not movies

**Reply by CherryFloors** [reply]
If you name this new library something like "Trailers" it may be clear enough. The plugin will search all libraries that hold "Movies" for trailers so these mock movies don't need to be in the same library as your actual movies. I don't know if that helps.
